src/song_loader.cpp

```diff
@@ -0,0 +1,98 @@
+#include "song_loader.h"
+
+SongLoader::SongLoader()
+{
+    std::string dir = std::string(".");
+    check_library = read_song_names(dir, files);
+    if(check_library != false)
+    {
+        shuffle_songs();
+    }  
+}
+
+void SongLoader::notify_listeners()
+{
+    for(I_SongLoaderObserver*listener:listeners) 
+    {
+        listener->current_song(std::string song_path);
+    }
+}
+
+void SongLoader::next_song()
+{
+    files[(current_song_index+1)%files.size()];
+    notify_listeners();
+}
+
+void SongLoader::previous_song()
+{
+    files[(current_song_index-1)%files.size()];
+    notify_listeners();
+}
+
+std::string SongLoader::next_song_name()
```

Comment:
I think you have `next_song_name()` and `next_song()` backwards. (same for previous_song)

your `next_song_name()` changes `current_song_index`, which it shouldn't, while `next_song()` does not change `current_song_index`